pytest_lab/locker.py

```diff
@@ -122,22 +122,31 @@ def __init__(self, config, backend, ttl=30):
         self._thread = None
         self._stop = threading.Event()
 
-    def aquire(self, key, user=None):
+    def aquire(self, key, wait_on_locks, user=None):
         record = self.backend.read(key)
 
-        if record and record.ttl:
-            logger.error('{} is locked by {}, waiting {} seconds for lock '
-                         'to expire...'.format(key, record.value, record.ttl + 1))
-            start = time.time()
-            while time.time() - start < record.ttl + 1:
-                record = self.backend.read(key)
-                if not record:
+        def check_lock(record):
+            if record and record.ttl:
+                logger.error(
+                    '{} is locked by {}, waiting {} seconds for lock '
+                    'to expire...'.format(key, record.value, record.ttl + 1))
+                start = time.time()
+                while time.time() - start < record.ttl + 1:
+                    record = self.backend.read(key)
+                    if not record:
+                        return True
+                    time.sleep(0.5)
+            else:
+                return True
+
+        # Check if the key is locked or not
+        if not check_lock(record):
+            if not wait_on_locks:
+                raise ResourceLocked(
```

Comment:
This doesn't make sense: throw an exception if its not locked, and we don't want to wait on locks?

Reply:
I think you're missing the point. This reads, to me as "**If not locked** and we're not waiting on locks, throw an error that we're locked".

Reply:
The check_lock returns True if there is no existing lock. Otherwise, it returns False if there is an existing lock.
The intended logic is "If locked and not waiting on lock, throw an error we are locked"

Reply:
You're missing the point - I've read the code and I understand what it does. That's not where I'm confused.

My point is, in isolation, if I see `if not check_lock(...)`, and I have no idea what `check_lock` does because I haven't read the function in depth yet - would I be more likely to assume that it checks if something is currently locked? Or would I be more likely to assume it checks that something isn't locked.

How does it read as English to you?

There's no negative in the function name to make me think you're going to return true if we don't have a lock. And - if that was the case - its weird to make this constructed with a double negative.